### PR TITLE
NumericInput: refactor and extend spinner

### PR DESCRIPTION
**Pull request by driesd**
### Description

Before this PR it was only possible to render de `spinner controls` inside the input field, on the left. This PR not only renames the `spinner` prop to `stepper`, but also introduces a new way of rendering these controls.

#### No stepper at all
Screenshot 2019-12-17 16 13 12

#### Stepper as a suffix
Screenshot 2019-12-17 16 12 54

#### Stepper as two connected buttons (new)
Screenshot 2019-12-17 16 13 04

### Breaking changes

- changed prop type `spinner` from `boolean` (default `true`) to `oneOf` (default `suffix`)
- renamed `spinner` prop to `stepper`

### CODE REVIEW

File: src/components/input/NumericInput.js [hunk 1]
```diff
@@ -89,35 +96,80 @@ class NumericInput extends PureComponent {
...
+        <Button
+          disabled={this.isMaxReached()}
+          icon={<IconAddSmallOutline />}
+          onClick={this.handleIncreaseValue}
+          size={size}
+        />
+      );
+    }
+
+    return connectedRight;
```
Comment (sanderbrugge) [inline src/components/input/NumericInput.js]: Q: not entirely sure what this returns? wouldn't `null` suffice?

File: src/components/input/NumericInput.js [hunk 2]
```diff
@@ -89,35 +96,80 @@ class NumericInput extends PureComponent {
...
+        <Button
+          disabled={this.isMinReached()}
+          icon={<IconMinusSmallOutline />}
+          onClick={this.handleDecreaseValue}
+          size={size}
+        />
+      );
+    }
+
+    return connectedLeft;
```
Comment (sanderbrugge) [inline src/components/input/NumericInput.js]: Same remark as `getConnectedRight`.

File: src/components/input/NumericInput.js (hunk 1 above)
Comment (driesd) [reply src/components/input/NumericInput.js]: A: `connectedRight` and `connectedLeft` are props that can contain f.e. a `Button`. But, when the stepper controls are rendered left and right, we don't want any extra connected component being rendered.

Comment (rathesDot) [reply src/components/input/NumericInput.js]: Simply speaking, if we have a `<NumericInput stepper="connected" connectedRight={<Button />}>` the `connectedRight` would be overwritten.

Comment (sanderbrugge) [reply src/components/input/NumericInput.js]: @driesd @rathesDot thanks for explaining, missed that context. Still feels like an odd architecture though.

**Pull request merged**